sdks/java/io/hdfs/pom.xml

```diff
@@ -58,12 +58,41 @@
     </dependency>
 
     <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-annotations</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+    </dependency>
+
+    <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-client</artifactId>
       <version>2.7.0</version>
       <scope>provided</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
+      <version>2.7.0</version>
```

Comment:
For question 1:
I don't know why hadoop-client 2.7.0 is picked at the very beginning.
The new hadoop dependencies I added was previously pulled from hadoop-client. So, the dependency plugin suggests 2.7.0.

I agree HadoopIO should work with all runners ideally.
To track this issue, I filed https://issues.apache.org/jira/browse/BEAM-375

The version conflicts already exists, so I propose to address it separate from this PR.
It can be as easy as changing to 2.2.0 if HadoopIO works with it.